lib/inspec/requirement.rb

```diff
@@ -16,15 +20,66 @@ def matches_spec?(spec)
       @dep.match?(params[:name], params[:version])
     end
 
+    def source_url
+      case source_type
+      when :local_path
+        "file://#{File.expand_path(opts[:path])}"
+      when :url
+        @opts[:url]
+      end
+    end
+
+    def local_path
+      @local_path ||= case source_type
+                      when :local_path
+                        File.expand_path(opts[:path])
+                      else
+                        @vendor_index.prefered_entry_for(@name, source_url)
+                      end
+    end
+
+    def source_type
+      @source_type ||= if @opts[:path]
+                         :local_path
+                       elsif opts[:url]
+                         :url
+                       else
+                         fail "Cannot determine source type from #{opts}"
+                       end
+    end
+
+    def fetcher_class
+      @fetcher_class ||= case source_type
```

Comment:
Let me know if you think we should merge those implementations in another iteration.